datafusion/sqllogictest/test_files/tpch/q22.slt.part

```diff
@@ -82,27 +82,28 @@ SortPreservingMergeExec: [cntrycode@0 ASC NULLS LAST]
 ----------RepartitionExec: partitioning=Hash([cntrycode@0], 4), input_partitions=4
 ------------AggregateExec: mode=Partial, gby=[cntrycode@0 as cntrycode], aggr=[COUNT(*), SUM(custsale.c_acctbal)]
 --------------ProjectionExec: expr=[substr(c_phone@0, 1, 2) as cntrycode, c_acctbal@1 as c_acctbal]
-----------------NestedLoopJoinExec: join_type=Inner, filter=CAST(c_acctbal@0 AS Decimal128(19, 6)) > AVG(customer.c_acctbal)@1
-------------------CoalesceBatchesExec: target_batch_size=8192
---------------------HashJoinExec: mode=Partitioned, join_type=LeftAnti, on=[(c_custkey@0, o_custkey@0)], projection=[c_phone@1, c_acctbal@2]
+----------------RepartitionExec: partitioning=RoundRobinBatch(4), input_partitions=1
+------------------NestedLoopJoinExec: join_type=Inner, filter=CAST(c_acctbal@0 AS Decimal128(19, 6)) > AVG(customer.c_acctbal)@1
+--------------------CoalescePartitionsExec
```

Comment:
Here as well. before we had no `CoalescePartitionsExec` here so from the plan it looks like this join was using more parallelism before?

Reply:
Same as above -- the reason is that currently result of final aggregation (estimated as single row), cannot be reordered with result of partitioned aggregation (abscent statistics)